### PR TITLE
Adds dispose to TextEditingController

### PR DESCRIPTION
**Pull request by vicenterusso**
Fixes https://github.com/joanpablo/reactive_forms/issues/445

### CODE REVIEW

File: lib/src/widgets/reactive_text_field.dart
Comment (remonke) [inline lib/src/widgets/reactive_text_field.dart:1]: I suggest you only dispose only if the controller is not provided in the widget.

Just simply add
```dart
if (widget.textController == null) {
  //
}
```

Comment (vicenterusso) [reply lib/src/widgets/reactive_text_field.dart:1]: At this point, inside `_ReactiveTextFieldState` class, `textController` is not nullable

Comment (remonke) [reply lib/src/widgets/reactive_text_field.dart:1]: `widget` refers the Widget, ya know, this thing with `StatefulWidget` class, not `State`. In `initState` you either use `textController` from this widget or initialize it. I suggest you only dispose the one initialized in the state, and if it's externally provided, leave it alone.

Comment (vicenterusso) [reply lib/src/widgets/reactive_text_field.dart:1]: That makes total sense!

**Comment by joanpablo** [comment]
Hi @vicenterusso, thanks for the contribution.
It seems to be an issue with the patch though, can you check what is failing before I can take a look? Thanks in advance.

**Pull request merged**